lttoolbox/transducer.cc

```diff
@@ -319,18 +319,18 @@ Transducer::determinize(int const epsilon_tag)
 
   int t = 0;
 
+  set<int> finals_state;
+  for(auto& it2 : finals) {
+	finals_state.insert(it2.first);
+  }
+
```

Comment:
for our edification, can you explain why this block is moved?

Reply:
It's calculating a list of the final states in the original transducer, but in its original location it was recalculating it on each iteration but it won't have changed.